File: spark-event-gateway-srv/src/main/java/fr/an/spark/gateway/dto/BlockManagerId.java
package fr.an.spark.gateway.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

public class BlockManagerId {
    
    @JsonProperty("Executor ID")
    public String executorId;
    
    @JsonProperty("Host")
    public String host;
    
    @JsonProperty("Port")
    public int port;
}

File: spark-event-ui/src/app/model/sparkevents/BlockManagerId.ts

export class BlockManagerId {
    
  // @JsonProperty("Executor ID")
  readonly executorId: string;
  
  // @JsonProperty("Host")
  readonly host: string;
  
  // @JsonProperty("Port")
  readonly port: number;

  constructor(executorId: string, host: string, port: number) {
    this.executorId = executorId;
    this.host = host;
    this.port = port;
  }
  
  static fromJson(src: any): BlockManagerId {
    let executorId = <string> src['Executor ID'];
    let host = <string> src['Host'];
    let port = <number> src['Port'];
    return new BlockManagerId(executorId, host, port);
  }

  static createDefault(): BlockManagerId {
    return new BlockManagerId('', '', 0);
  }

}
